frontend/apps/app/app/(app)/app/projects/[projectId]/layout.tsx

```diff
@@ -0,0 +1,92 @@
+import type { LayoutProps } from '@/app/types'
+import { TabsContent, TabsRoot } from '@liam-hq/ui'
+import { headers } from 'next/headers'
+import { notFound } from 'next/navigation'
+import { safeParse } from 'valibot'
+import * as v from 'valibot'
+import { ProjectHeader } from './components/ProjectHeader'
+import {
+  PROJECT_TAB,
+  ProjectTabSchema,
+  type ProjectTabValue,
+} from './components/ProjectHeader/projectConstants'
+import styles from './layout.module.css'
+
+const paramsSchema = v.object({
+  projectId: v.string(),
+})
+
+const getDefaultTabFromPath = async (): Promise<
+  ProjectTabValue | undefined
+> => {
+  const headersList = await headers()
+  const urlPath = headersList.get('x-url-path') || ''
+  const pathSegments = urlPath.split('/')
+  const lastSegment = pathSegments[pathSegments.length - 1]
+
+  const result = safeParse(ProjectTabSchema, lastSegment)
+
+  return result.success ? result.output : undefined
+}
+
+export default async function ProjectLayout({ children, params }: LayoutProps) {
+  const defaultTabFromPath = await getDefaultTabFromPath()
+  const parsedParams = v.safeParse(paramsSchema, await params)
+  if (!parsedParams.success) return notFound()
+
+  // Extract branchOrCommit from the URL path
+  const headersList = await headers()
+  const urlPath = headersList.get('x-url-path') || ''
+  const pathSegments = urlPath.split('/')
+
+  // Find the index of 'ref' in the path and get the next segment as branchOrCommit
+  const refIndex = pathSegments.findIndex(
+    (segment: string) => segment === 'ref',
+  )
+  const branchOrCommit =
+    refIndex !== -1 && pathSegments.length > refIndex + 1
+      ? pathSegments[refIndex + 1]
+      : 'main' // TODO: get default branch from API(using currentOrganization)
+
+  return (
+    <div className={styles.container}>
+      <div className={styles.contentContainer}>
+        <h1 className={styles.heading}>Project</h1>
```

Comment:
I think it would be good to merge them once, but there is no `Projects` string in the design and the header is fixed at the top.
It would be better to have this fixed together.

<img width="501" alt="Screenshot 2025-04-24 at 14 05 28" src="https://github.com/user-attachments/assets/94e013f3-5ec9-4cf5-b40e-d258b34bc772" />

Reply:
Oh!Thanks!🙏
I will fix it on this PR!

Reply:
I fixed it and will merge this PR!
https://github.com/liam-hq/liam/pull/1456/commits/c8f07f6de32dbf76b7adb3f27195d9acb9d049fd

![ss 3188](https://github.com/user-attachments/assets/6a134e0e-b01d-429a-b5ce-3f1213ef8df3)

Reply:
Thanks 👍🏻